### DISCUSSION TITLE
Dynamically Set The Return type of a query

### DISCUSSION THREAD

**Discussion by EdgarEmmanuel** [rebing/graphql-laravel#807] [Q&A]
Hello everybody , it's been four days i am trying do do something and i cam across the fact that does not exist yet so this is waht i want 

what i want to do is something like that

query { users(Paginate:true,count:2,page:1){ metadata{ per_page, total }, data{ date } } }

it returns me this :

{ "data": { "users": { "metadata": { "per_page": 2 }, "data": [ { "date": "2021-14-07" } ] } } }

==========================

and when i only do this

query{ users(Paginated:false){ date } }

it returns me

{ "data": { "users": [ { "date": "2021-14-07" } ] } }

is this possible in graphQL ? and if yes is this possible how to implements is with This package ?

**Comment by mfn** [comment]
Hi there, I think you also asked on slack [where I already replied](https://rebing-graphql.slack.com/archives/CF9LP4W2F/p1626707744003300), but I'll replicate it here too

The TL;DR is that I think you can do this with unions:
- you define your query to return a union type, compromising of the types you want to return
- you then need to implement the union and implement the discriminator when to return which type (e.g. what kind of shape/payload)

Notion that, the way this works in GraphQL is that you need to explicitly formulate all possible return types in the query:
```graphql
query foo(arg1:true) {
  ...resultTypeWithPagination {
    metadata {
    }
    data {
    }
  }
  ...resultTypeWithoutPagination {
    user
  }
}
```
HTH!
